src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFIll.Fixer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1849: Use Span.Clear instead of Span.Fill(default)
+    /// </summary>
+    public abstract class UseSpanClearInsteadOfFillFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; }
+            = ImmutableArray.Create(UseSpanClearInsteadOfFillAnalyzer.DiagnosticId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            var invocationTarget = GetInvocationTarget(node);
+            if (invocationTarget == null)
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics[0];
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillCodeFixTitle,
+                    createChangedDocument: async cancellationToken =>
+                    {
+                        DocumentEditor editor = await DocumentEditor.CreateAsync(context.Document, cancellationToken).ConfigureAwait(false);
+                        SyntaxGenerator generator = editor.Generator;
+
+                        var memberAccess = generator.MemberAccessExpression(invocationTarget, UseSpanClearInsteadOfFillAnalyzer.ClearMethod);
+                        var invocation = generator.InvocationExpression(memberAccess);
+
+                        editor.ReplaceNode(node, invocation);
+                        return editor.GetChangedDocument();
```

Comment:
Instead of that, why not replace `GetInvocationTarget` with another method that returns `MemberAccessExpressionSyntax.Name` instead of `MemberAccessExpressionSyntax.Expression`, and just replace this node with `SyntaxFactory.IdentifierName("Clear")`

Reply:
I don't understand what you mean. The method and arguments both need to be updated, so updating the whole invocation expression is natural.

Reply:
Yeah my comment doesn't make sense at all 😄